### ISSUE TITLE
Segmentation fault when grepping

### ISSUE THREAD

**Issue by schra** [terminaldweller/cgrep#13]
What I observe:

```
$ ~/Git/cgrep/cgrep --var --regex ".*[a-z][A-Z].*" aerialmap_display.cpp
Segmentation fault
```

Also don't hit me, but with the debug symbol everything works :(

```
$ ~/Git/cgrep/cgrep-dbg --var --regex ".*[a-z][A-Z].*" aerialmap_display.cpp
/home/andre/Git/rviz_satellite/src/aerialmap_display.cpp:489:19:      TileId const toFind{ center_tile_->tile_server, { xx, yy }, center_tile_->zoom };
```

Backtrace (but without debug symbols as mentioned above):

```
$ lldb-10 /home/andre/.local/bin/cgrep
(lldb) target create "/home/andre/.local/bin/cgrep"
Current executable set to '/home/andre/.local/bin/cgrep' (x86_64).
(lldb) r --var --regex ".*[a-z][A-Z].*" aerialmap_display.cpp
Process 7022 launched: '/home/andre/.local/bin/cgrep' (x86_64)
Process 7022 stopped
* thread #1, name = 'cgrep', stop reason = signal SIGSEGV: invalid address (fault address: 0x108)
    frame #0: 0x00000000007ca20a cgrep`clang::SourceManager::getDecomposedExpansionLoc(clang::SourceLocation) const + 26
cgrep`clang::SourceManager::getDecomposedExpansionLoc:
->  0x7ca20a <+26>: movl   0x108(%rdi), %esi
    0x7ca210 <+32>: movl   %ebx, %edx
    0x7ca212 <+34>: callq  0x453990                  ; clang::SourceManager::isOffsetInFileID(clang::FileID, unsigned int) const
    0x7ca217 <+39>: testb  %al, %al
(lldb) bt
* thread #1, name = 'cgrep', stop reason = signal SIGSEGV: invalid address (fault address: 0x108)
  * frame #0: 0x00000000007ca20a cgrep`clang::SourceManager::getDecomposedExpansionLoc(clang::SourceLocation) const + 26
    frame #1: 0x00000000007cd653 cgrep`clang::SourceManager::getFileCharacteristic(clang::SourceLocation) const + 19
    frame #2: 0x00000000004b1395 cgrep`Devi::IsTheMatchInSysHeader(bool, clang::ast_matchers::MatchFinder::MatchResult const&, clang::SourceLocation) + 21
    frame #3: 0x000000000049c1ed cgrep`VDecl::run(clang::ast_matchers::MatchFinder::MatchResult const&) + 333
    frame #4: 0x00000000010662e1 cgrep`clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor::MatchVisitor::visitMatch(clang::ast_matchers::BoundNodes const&) + 193
    frame #5: 0x00000000010808ac cgrep`clang::ast_matchers::internal::BoundNodesTreeBuilder::visitMatches(clang::ast_matchers::internal::BoundNodesTreeBuilder::Visitor*) + 204
    frame #6: 0x0000000001065b62 cgrep`clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor::matchWithFilter(clang::ast_type_traits::DynTypedNode const&) + 1154
    frame #7: 0x000000000107c92b cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseTypeLoc(clang::TypeLoc) + 2795
    frame #8: 0x000000000108045b cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseFunctionHelper(clang::FunctionDecl*) + 667
    frame #9: 0x00000000010680c4 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 1748
    frame #10: 0x000000000106d1d7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #11: 0x0000000001067d8a cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 922
    frame #12: 0x000000000106b5fd cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 15373
    frame #13: 0x000000000106d1d7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #14: 0x0000000001067a8a cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 154
    frame #15: 0x000000000106d1d7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #16: 0x0000000001069efb cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 9483
    frame #17: 0x000000000104baf9 cgrep`clang::ast_matchers::MatchFinder::matchAST(clang::ASTContext&) + 825
    frame #18: 0x000000000049c066 cgrep`CgrepASTConsumer::HandleTranslationUnit(clang::ASTContext&) + 38
    frame #19: 0x0000000000934023 cgrep`clang::ParseAST(clang::Sema&, bool, bool) + 643
    frame #20: 0x00000000007d8e88 cgrep`clang::FrontendAction::Execute() + 72
    frame #21: 0x0000000000800921 cgrep`clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) + 1585
    frame #22: 0x00000000011b93bc cgrep`clang::tooling::FrontendActionFactory::runInvocation(std::shared_ptr<clang::CompilerInvocation>, clang::FileManager*, std::shared_ptr<clang::PCHContainerOperations>, clang::DiagnosticConsumer*) + 412
    frame #23: 0x00000000011b9126 cgrep`clang::tooling::ToolInvocation::runInvocation(char const*, clang::driver::Compilation*, std::shared_ptr<clang::CompilerInvocation>, std::shared_ptr<clang::PCHContainerOperations>) + 262
    frame #24: 0x00000000011b85bf cgrep`clang::tooling::ToolInvocation::run() + 2655
    frame #25: 0x00000000011ba7ea cgrep`clang::tooling::ClangTool::run(clang::tooling::ToolAction*) + 3002
    frame #26: 0x000000000043417b cgrep`main + 635
    frame #27: 0x00007ffff29d4b97 libc.so.6`__libc_start_main(main=(cgrep`main), argc=5, argv=0x00007fffffffb9d8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffb9c8) at libc-start.c:310
    frame #28: 0x00000000004328aa cgrep`_start + 42
```

I'm on Ubuntu and compiled cgrep like this:

```
$ make CXX=clang-10 LLVM_CONF=llvm-config-10 BUILD_MODE=DEBUG cgrep-dbg
clang-10 -fpic -I/usr/lib/llvm-10/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/usr/lib/llvm-10/build//tools/clang/include -I/usr/lib/llvm-10/build//tools/clang/include -std=c++17 -fexceptions  -g -c cgrep.cpp -o cgrep.odbg
clang-10 -fpic -I/usr/lib/llvm-10/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/usr/lib/llvm-10/build//tools/clang/include -I/usr/lib/llvm-10/build//tools/clang/include -std=c++17 -fexceptions  -g -c cfe-extra/cfe_extra.cpp -o cfe-extra/cfe_extra.odbg
clang-10 cgrep.odbg cfe-extra/cfe_extra.odbg  -Wl,--start-group -lclangAST -lclangAnalysis -lclangBasic -lclangDriver -lclangEdit -lclangFrontend -lclangFrontendTool -lclangLex -lclangParse -lclangSema -lclangEdit -lclangASTMatchers -lclangRewrite -lclangRewriteFrontend -lclangStaticAnalyzerFrontend -lclangStaticAnalyzerCheckers -lclangStaticAnalyzerCore -lclangSerialization -lclangToolingCore -lclangTooling -lstdc++ -lLLVMRuntimeDyld -lm -Wl,--end-group -L/usr/lib/llvm-10/lib  -lLLVM-10  -g -o cgrep-dbg

$ make CXX=clang-10 LLVM_CONF=llvm-config-10
clang-10 -fpic -I/usr/lib/llvm-10/include -std=c++14   -fno-exceptions -D_GNU_SOURCE -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -I/usr/lib/llvm-10/build//tools/clang/include -I/usr/lib/llvm-10/build//tools/clang/include -std=c++17 -fexceptions  -c cgrep.cpp -o cgrep.o
clang-10 cgrep.o cfe-extra/cfe_extra.o  -Wl,--start-group -lclangAST -lclangAnalysis -lclangBasic -lclangDriver -lclangEdit -lclangFrontend -lclangFrontendTool -lclangLex -lclangParse -lclangSema -lclangEdit -lclangASTMatchers -lclangRewrite -lclangRewriteFrontend -lclangStaticAnalyzerFrontend -lclangStaticAnalyzerCheckers -lclangStaticAnalyzerCore -lclangSerialization -lclangToolingCore -lclangTooling -lstdc++ -lLLVMRuntimeDyld -lm -Wl,--end-group -L/usr/lib/llvm-10/lib  -lLLVM-10  -o cgrep
```

Also if it helps: This is the repo where I tried to grep: https://github.com/gareth-cross/rviz_satellite

**Comment by terminaldweller** [comment]
I tried the example on ubuntu 20.04 with llvm-10. Cloned the master from the repo you mentioned and ran the same regex and it works fine:
```
➜  build git:(master) ✗ ~/devi/cgrep/cgrep --var --regex ".*[a-z][A-Z].*" ../src/aerialmap_display.cpp
../src/aerialmap_display.cpp:389:13:  auto const tileCoordinates = fromWGSCoordinate({ msg->latitude, msg->longitude }, zoom_);
../src/aerialmap_display.cpp:390:15:  TileId const newCenterTileID{ tile_url_, tileCoordinates, zoom_ };
../src/aerialmap_display.cpp:391:13:  bool const centerTileChanged = (!lastCenterTile_ || !(newCenterTileID == *lastCenterTile_));
../src/aerialmap_display.cpp:441:14:  float const errorRate = tileCache_.getTileServerErrorRate(tile_url_);
../src/aerialmap_display.cpp:476:7:  bool loadedAllTiles = true;
../src/aerialmap_display.cpp:488:19:      TileId const toFind{ lastCenterTile_->tileServer, { xx, yy }, lastCenterTile_->zoom };
../src/aerialmap_display.cpp:647:13:  bool const gotTransform =
../src/aerialmap_display.cpp:655:13:  auto const centerTile = fromWGSCoordinate<double>({ ref_fix_->latitude, ref_fix_->longitude }, zoom_);
../src/aerialmap_display.cpp:660:13:  auto const centerTileOffsetX = centerTile.x - std::floor(centerTile.x);
../src/aerialmap_display.cpp:663:13:  auto const centerTileOffsetY = 1 - (centerTile.y - std::floor(centerTile.y));
../src/aerialmap_display.cpp:668:13:  auto const translationAerialMapToNavSatFix =
../src/aerialmap_display.cpp:670:13:  auto const translationNavSatFixToAerialMap = -translationAerialMapToNavSatFix;
```
I guess as a next step I should replicate exactly your setup so please let me know which version of Ubuntu you're one and if you got llvm-10 from repo or apt.llvm.org or any other source.
Cheers!
P.S.:Also which branch of cgrep are you running?

**Comment by schra** [comment]
Soo, this is weird. I tried to reproduce the segmentation fault via Docker by installing the same exact versions of the llvm/ clang packages and by creating the same environment as on my Ubuntu PC. However, in Docker everything runs smooth. Any ideas how this can be further debugged, if it works fine in the Docker container?

The missing specs you asked for:
* The PPA is oibaf/graphics-drivers
* The OS is Ubuntu Bionic (not 20.04)
* The branch is the PR https://github.com/gareth-cross/rviz_satellite/pull/79

Here is my Dockerfile where everything works:

```
# ros:melodic is a Dockefile that helps us to install dependencies for
# rviz_satellite. it is based on ubuntu:bionic
FROM ros:melodic

# we need bash, so that we can source, which is required by ROS
SHELL ["/bin/bash", "-c"]

# build rviz_satellite
RUN apt-get update \
 && apt-get install -y --no-install-recommends git make ca-certificates python-catkin-tools \
 && mkdir -p /catkin_ws/src && cd /catkin_ws/src \
 && git clone https://github.com/gareth-cross/rviz_satellite \
 && cd rviz_satellite/ \
 && git fetch origin pull/79/head:cleanup && git checkout cleanup \
 && cd /catkin_ws \
 && source /opt/ros/melodic/setup.bash \
 && rosdep install -iry --from-path src \
 && catkin build -DCMAKE_EXPORT_COMPILE_COMMANDS=1 \
 && source devel/setup.bash \
 && cp build/rviz_satellite/compile_commands.json src/rviz_satellite/ \
 && cd src/rviz_satellite/

# clang/ llvm version
ENV X 10

# install llvm/clang
RUN apt-get update \
 && apt-get install -y --no-install-recommends software-properties-common \
 && add-apt-repository ppa:oibaf/graphics-drivers \
 && apt-get install -y --no-install-recommends clang-$X llvm-$X-dev libclang-common-$X-dev libclang-$X-dev

# build cgrep
RUN git clone https://github.com/bloodstalker/cgrep \
 && cd cgrep \
 && git submodule init \
 && git submodule update \
 && make CXX=clang-$X LLVM_CONF=llvm-config-$X BUILD_MODE=DEBUG cgrep-dbg \
 && make CXX=clang-$X LLVM_CONF=llvm-config-$X
```

**Comment by terminaldweller** [comment]
One suspicion I had was that the problem is with LLVM/Clang itself. The LLVM on your Ubuntu machine is not really the exact same as the one on the docker image.
But looking though oibaf's PPA, the last update time for the LLVM toolchain on there is 2020-04-19.
So I guess it can't be it.
Another angle is just go with what the lldb call stack tells us which is the `SourceLocation` passed is invalid. To check if that's the case, I made a new branch. It's called `mystery`, Please build that branch and let me know what happens.

**Comment by schra** [comment]
> One suspicion I had was that the problem is with LLVM/Clang itself. The LLVM on your Ubuntu machine is not really the exact same as the one on the docker image.
> But looking though oibaf's PPA, the last update time for the LLVM toolchain on there is 2020-04-19.
So I guess it can't be it.

You are correct, the package versions are identical:

Inside the Docker container:

```
# apt-cache policy clang-$X llvm-$X-dev libclang-common-$X-dev libclang-$X-dev
clang-10:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
llvm-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
libclang-common-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
libclang-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
```

In my Ubuntu PC:

```
$ apt-cache policy clang-$X llvm-$X-dev libclang-common-$X-dev libclang-$X-dev
clang-10:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
llvm-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
libclang-common-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
libclang-10-dev:
  Installed: 1:10.0.0-4~oibaf~b
  Candidate: 1:10.0.0-4~oibaf~b
  Version table:
 *** 1:10.0.0-4~oibaf~b 500
        500 http://ppa.launchpad.net/oibaf/graphics-drivers/ubuntu bionic/main amd64 Packages
        100 /var/lib/dpkg/status
```

>  Another angle is just go with what the lldb call stack tells us which is the SourceLocation passed is invalid. To check if that's the case, I made a new branch. It's called mystery, Please build that branch and let me know what happens.

Not exactly sure what output you want from me, but I build the `mystery` branch and here is what happened:

```
$ (cd ~/Git/cgrep; git branch)
  master
* mystery

$ ~/Git/cgrep/cgrep --var --regex ".*[a-z][A-Z].*" aerialmap_display.cpp
Segmentation fault

$ lldb-10 ~/Git/cgrep/cgrep
(lldb) target create "/home/andre/Git/cgrep/cgrep"
Current executable set to '/home/andre/Git/cgrep/cgrep' (x86_64).
(lldb) r --var --regex ".*[a-z][A-Z].*" aerialmap_display.cpp
Process 22458 launched: '/home/andre/Git/cgrep/cgrep' (x86_64)
Process 22458 stopped
* thread #1, name = 'cgrep', stop reason = signal SIGSEGV: invalid address (fault address: 0x108)
    frame #0: 0x00000000007ca22a cgrep`clang::SourceManager::getDecomposedExpansionLoc(clang::SourceLocation) const + 26
cgrep`clang::SourceManager::getDecomposedExpansionLoc:
->  0x7ca22a <+26>: movl   0x108(%rdi), %esi
    0x7ca230 <+32>: movl   %ebx, %edx
    0x7ca232 <+34>: callq  0x453990                  ; clang::SourceManager::isOffsetInFileID(clang::FileID, unsigned int) const
    0x7ca237 <+39>: testb  %al, %al
(lldb) bt
* thread #1, name = 'cgrep', stop reason = signal SIGSEGV: invalid address (fault address: 0x108)
  * frame #0: 0x00000000007ca22a cgrep`clang::SourceManager::getDecomposedExpansionLoc(clang::SourceLocation) const + 26
    frame #1: 0x00000000007cd673 cgrep`clang::SourceManager::getFileCharacteristic(clang::SourceLocation) const + 19
    frame #2: 0x00000000004b13b5 cgrep`Devi::IsTheMatchInSysHeader(bool, clang::ast_matchers::MatchFinder::MatchResult const&, clang::SourceLocation) + 21
    frame #3: 0x000000000049c231 cgrep`VDecl::run(clang::ast_matchers::MatchFinder::MatchResult const&) + 401
    frame #4: 0x0000000001066301 cgrep`clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor::MatchVisitor::visitMatch(clang::ast_matchers::BoundNodes const&) + 193
    frame #5: 0x00000000010808cc cgrep`clang::ast_matchers::internal::BoundNodesTreeBuilder::visitMatches(clang::ast_matchers::internal::BoundNodesTreeBuilder::Visitor*) + 204
    frame #6: 0x0000000001065b82 cgrep`clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor::matchWithFilter(clang::ast_type_traits::DynTypedNode const&) + 1154
    frame #7: 0x000000000107c94b cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseTypeLoc(clang::TypeLoc) + 2795
    frame #8: 0x000000000108047b cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseFunctionHelper(clang::FunctionDecl*) + 667
    frame #9: 0x00000000010680e4 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 1748
    frame #10: 0x000000000106d1f7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #11: 0x0000000001067daa cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 922
    frame #12: 0x000000000106b61d cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 15373
    frame #13: 0x000000000106d1f7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #14: 0x0000000001067aaa cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 154
    frame #15: 0x000000000106d1f7 cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDeclContextHelper(clang::DeclContext*) + 87
    frame #16: 0x0000000001069f1b cgrep`clang::RecursiveASTVisitor<clang::ast_matchers::internal::(anonymous namespace)::MatchASTVisitor>::TraverseDecl(clang::Decl*) + 9483
    frame #17: 0x000000000104bb19 cgrep`clang::ast_matchers::MatchFinder::matchAST(clang::ASTContext&) + 825
    frame #18: 0x000000000049c066 cgrep`CgrepASTConsumer::HandleTranslationUnit(clang::ASTContext&) + 38
    frame #19: 0x0000000000934043 cgrep`clang::ParseAST(clang::Sema&, bool, bool) + 643
    frame #20: 0x00000000007d8ea8 cgrep`clang::FrontendAction::Execute() + 72
    frame #21: 0x0000000000800941 cgrep`clang::CompilerInstance::ExecuteAction(clang::FrontendAction&) + 1585
    frame #22: 0x00000000011b93dc cgrep`clang::tooling::FrontendActionFactory::runInvocation(std::shared_ptr<clang::CompilerInvocation>, clang::FileManager*, std::shared_ptr<clang::PCHContainerOperations>, clang::DiagnosticConsumer*) + 412
    frame #23: 0x00000000011b9146 cgrep`clang::tooling::ToolInvocation::runInvocation(char const*, clang::driver::Compilation*, std::shared_ptr<clang::CompilerInvocation>, std::shared_ptr<clang::PCHContainerOperations>) + 262
    frame #24: 0x00000000011b85df cgrep`clang::tooling::ToolInvocation::run() + 2655
    frame #25: 0x00000000011ba80a cgrep`clang::tooling::ClangTool::run(clang::tooling::ToolAction*) + 3002
    frame #26: 0x000000000043417b cgrep`main + 635
    frame #27: 0x00007ffff29d4b97 libc.so.6`__libc_start_main(main=(cgrep`main), argc=5, argv=0x00007fffffffb9d8, init=<unavailable>, fini=<unavailable>, rtld_fini=<unavailable>, stack_end=0x00007fffffffb9c8) at libc-start.c:310
    frame #28: 0x00000000004328aa cgrep`_start + 42
```

**Comment by terminaldweller** [comment]
First, a little explanation on the `mystery` branch:
I added an extra line before where we failed, to check the `SourceLocation` for being valid before calling the function that eventually causes the crash to see if that helps but based on your result it didn't do anything.
I just tried your exact setup. Ubuntu 18.04, got LLVM 10 from oibaf's PPA, checked out PR-79 for rviz_sattelite and ran `cgrep --var --regex ".*[a-z][A-Z].*"` and it doesn't segfault.
I guess at this point I can just ask the clang maillist and see if anything comes out of that.
